vllm/model_executor/layers/attention.py

```diff
@@ -280,3 +286,25 @@ def _paged_attention(
             alibi_slopes,
         )
     return output
+
+
+def _multi_query_cached_kv_attention(query: torch.Tensor, key: torch.Tensor,
+                                     value: torch.Tensor,
+                                     key_cache: torch.Tensor,
+                                     value_cache: torch.Tensor,
+                                     input_metadata: InputMetadata):
+    # Pre-allocate the output tensor.
+    output = torch.empty_like(query)
+    context_attention_fwd(
+        query,
+        key,
+        value,
+        output,
+        key_cache,
+        value_cache,
+        input_metadata.block_tables,  # [BS, max_block_per_request]
+        input_metadata.start_loc,
+        input_metadata.context_lens,
+        input_metadata.sd_prompt_lens,
```

Comment:
Is the order of context_lens and sd_prompt_lens wrong? 
Do these two parameters correspond to the context_attention_fwd method parameters b_ctx_len and b_ctx_len?

    def context_attention_fwd(q, k, v, o, k_cache, v_cache, b_loc, b_start_loc,
                              b_seq_len, b_ctx_len, max_input_len):